lib/docker/imageservice.go

```diff
@@ -295,6 +310,86 @@ func (r *imageService) Sync(ctx context.Context, dir string, progress utils.Prin
 	return installedTags, nil
 }
 
+// SyncForScanning syncs all images in a local directory, with a remote registry that is capable of scanning and
+// reporting vulnerabilities in those images. All images and tags are written into a designated repository, with the
+// image tag rewritten to help identify the image.
+// Example of gravity 7.0.0:
+//   quay.io/gravitational/debian-tall:0.0.1 -> quay.io/gravitational/gravity-scan:7.0.0_gravitational_debian-tall_0.0.1
```

Comment:
If I'm not mistaken, the distribution name reference format allows [multiple name components](https://github.com/docker/distribution/blob/v2.7.1/reference/regexp.go#L53) so it could be possible to wrap the scan image under the scan sub-group:
```
quay.io/gravitational/gravity-scan/7.0.0/debian-tall:0.0.1
```
Could be easier to work with.

Reply:
While allowed by standard, my understanding is most public registries only allow one layer to the naming scheme. So <base>/<org>/<image> (quay.io/gravitational/example) so we would only be able to support this if we were using our own registry.